### PR TITLE
feat: AbortSignal cancellation for RPC, wired to async-render supersede (#492)

### LINKED ISSUE #492: feat: AbortSignal cancellation for RPC, wired to async-render supersede
Sub-issue of #488. webjs's RPC stub has no cancellation: a superseded `async render()` refetch keeps its network request in flight, and there is no way to abort an action call. TanStack Start passes a `signal: AbortSignal` to the server fn and the client can abort.

## Design / approach

Pass an `AbortSignal` to the action (the server handler aborts when the client disconnects) and expose cancellation to the client stub. Crucially, wire it to webjs's existing async-render supersede token (`__renderToken` in component.js): when a prop change supersedes an in-flight `async render()`, abort the previous render's RPC instead of letting it complete and get dropped. Compose with streaming RPC (a cancelled stream stops on both ends). Keep it opt-in / automatic where safe (a superseded render auto-aborts).

## Acceptance criteria

- [ ] The action receives an `AbortSignal`; aborting it (client disconnect or explicit) stops server work where possible.
- [ ] A superseded `async render()` refetch aborts its in-flight RPC (network probe shows the request cancelled).
- [ ] Composes with streaming RPC (a cancelled stream stops on both ends).
- [ ] Tests: unit + browser/e2e (cancel-on-supersede network probe).

### PR DESCRIPTION
**Pull request by vivek7405**
## Summary

Closes #492. Cancellation for server actions, wired to the async-render supersede.

**Server.** The RPC endpoint runs an action inside `runWithActionSignal(req.signal, () => fn(...args))`, so an action reads the request's `AbortSignal` via `actionSignal()` (exported from `@webjsdev/server`) and can stop expensive work (a DB query, an upstream fetch) when the client disconnects or aborts. Outside an action `actionSignal()` returns a never-aborting signal, so the same `fetch(url, { signal: actionSignal() })` line is safe on a direct server-to-server call.

**Client.** Each `async render()` binds a fresh `AbortController` as the active signal; the generated RPC stub captures it SYNCHRONOUSLY at call entry (before its first `await`, since the stub awaits `stringify` first) and binds the `fetch` to it. When the next render supersedes the previous one, the previous controller aborts, cancelling the in-flight fetch. The aborted superseded render's rejection is already dropped by the render-token guard in `_commitAsync`, so it surfaces no error. Best-effort by design: it binds for an action called in the synchronous window of `render()` (the common `const u = await getUser(id)` first statement); a later-`await` action falls back to the existing drop-on-supersede, exactly as before.

## Test plan

- [x] Server unit: `actionSignal` is never-aborting outside an action; `runWithActionSignal` exposes the signal; an aborted request signal is visible.
- [x] Server integration: the action sees `req.signal` (aborted state) through `invokeAction`.
- [x] Client unit: the active-signal holder.
- [x] Stub content: the fetch captures `const sig = __sig()` synchronously and binds `signal: sig`.
- [x] e2e: a superseded `<abort-demo>` render's in-flight slow GET is `net::ERR_ABORTED`.
- [x] Full node suite green (2475); the #469/#472/#488 e2e re-verified (no regression from the component.js change).

## Docs

AGENTS.md (server-actions surface) + the data-fetching docs page document `actionSignal()` and the auto-abort-on-supersede.

## Dogfood

website / docs / ui-website boot 200 in prod mode, no broken preloads; the blog e2e covers the new `/abort` fixture.

### CODE REVIEW

**Review by vivek7405 — COMMENTED** [review]
Sharp pass on edge cases and test validity. The e2e abort test is a real counterfactual (without the feature the superseded fetch completes, no ERR_ABORTED), the server integration test reaches the action with a buffered body, and the GET-cache + signal and seed + allocation interactions are fine. Two gaps, both fixed in [`75207243`](https://github.com/webjsdev/webjs/commit/75207243): the expose() REST path was not wired into actionSignal() (only the RPC path was), and the component abort wiring had no guard runnable by 'npm test' / the Browser CI job (only the WEBJS_E2E e2e). Added the REST wrap + test and a wtr browser counterfactual.

File: packages/server/src/actions.js [hunk 1]
```diff
@@ -722,7 +730,9 @@ export async function invokeExposedAction(idx, route, params, req, onError) {
   const fn = mod[route.fnName];
   if (typeof fn !== 'function') return new Response(`Unknown action ${route.fnName}`, { status: 404 });
   try {
-    const result = await fn(arg, { req, params });
+    // The REST boundary wires the request signal into actionSignal() too (#492),
+    // symmetric with the RPC path and the #245 validation contract.
```
Comment (vivek7405) [inline packages/server/src/actions.js:734]: The RPC path got runWithActionSignal but invokeExposedAction (the expose REST boundary) called fn directly, so an expose()d action read the never-aborting signal and could never see a client disconnect. Symmetric wrap added.

Comment (vivek7405) [reply packages/server/src/actions.js:734]: Fixed in [`75207243`](https://github.com/webjsdev/webjs/commit/75207243).

**Review by vivek7405 — COMMENTED** [review]
Final pass: the REST-path wiring and its test hold up (reverting the wrap reds the REST test), the browser counterfactual is deterministic (the abort fires synchronously in the superseding render, and reverting the .abort() reds exactly that test), all four new exports are declared, and the abort-rejection is swallowed by the existing token guard. Clean and merge-ready.
